inversion_sim.py

```diff
@@ -84,21 +137,36 @@ def update_seen(self):
             this_edge = tuple(sorted([self.gene_list[i],
                                       self.gene_list[i+1]]))
             if this_edge not in self.seen:
-                #if "B.1" in this_edge:
-                #    print(this_edge)
-                #if "A.1" in this_edge:
-                #    print(this_edge)
-                # add the edge to the seen graph
                 self.seen[this_edge] = self.cycle
                 # update the trace structure
                 for j in [0,1]:
                     other = 1 if j == 0 else 0
                     self.trace[this_edge[j]][-1] += 1
                     if this_edge[j].startswith("A") and this_edge[other].startswith("B"):
                         self.trace_AtoB[this_edge[j]][-1] += 1
+                        if self.trace_AtoB[this_edge[j]][-1] == self.genesB-(1 if this_edge[j].split('.')[1] == '1' else 0):
```

Comment:
Can you please comment these lines, in English, and provide a breakdown of how the enclosed statement is working? Just document it better since this is dense!